src/libs/ReportUtils.ts

```diff
@@ -3464,14 +3477,63 @@ function shouldHideReport(report: OnyxEntry<Report>, currentReportId: string): b
     return parentReport?.reportID !== report?.reportID && !isChildReportHasComment;
 }
 
+function transactionHasViolation(transactionID: string, transactionViolations: TransactionViolations): boolean {
+    const violations: TransactionViolation[] = transactionViolations[transactionID];
+    return violations.some((violation: TransactionViolation) => violation.type === 'violation');
+}
+
+/**
+ * Checks to see if a report's parentAction is a money request that contains a violation
+ */
+function transactionThreadHasViolations(report: Report, transactionViolations: TransactionViolations, reportActions: OnyxCollection<ReportActions> | null = {}): boolean {
+    if (!reportActions) {
+        return false;
+    }
+    const resolvedParentReportAction = reportActions[`${report.parentReportID}`]?.[`${report.parentReportActionID}`];
+    if (!resolvedParentReportAction) {
+        return false;
+    }
+    if (resolvedParentReportAction.actionName !== CONST.REPORT.ACTIONS.TYPE.IOU) {
+        return false;
+    }
+    const {IOUTransactionID, IOUReportID} = resolvedParentReportAction?.originalMessage;
+    if (!IOUTransactionID || !IOUReportID) {
+        return false;
+    }
+    if (!isCurrentUserSubmitter(IOUReportID)) {
+        return false;
+    }
+    if (report?.stateNum !== CONST.REPORT.STATE_NUM.OPEN && report?.stateNum !== CONST.REPORT.STATE_NUM.PROCESSING) {
+        return false;
+    }
+    return transactionHasViolation(IOUTransactionID, transactionViolations);
+}
+
+/**
+ * Checks to see if a report contains a violation
+ */
+function reportHasViolations(reportID: string, transactionViolations: TransactionViolations): boolean {
```

Comment:
Same here. `hasReportViolations`

Reply:
If we update `ReportUtils.transactionHasViolation(transactionID: string, transactionViolations: TransactionViolations): boolean` to `TransactionUtils.hasViolations(transactionID: string, transactionViolations: TransactionViolations): boolean`, we can make this one `ReportUtils.hasViolations`

Reply:
Renamed as requested.